pkg/volume/flexvolume/plugin.go

```diff
@@ -287,3 +291,38 @@ func (plugin *flexVolumePlugin) getDeviceMountPath(spec *volume.Spec) (string, e
 	mountsDir := path.Join(plugin.host.GetPluginDir(flexVolumePluginName), plugin.driverName, "mounts")
 	return path.Join(mountsDir, volumeName), nil
 }
+
+func (plugin *flexVolumePlugin) ExpandVolumeDevice(spec *volume.Spec, newSize resource.Quantity, oldSize resource.Quantity) (resource.Quantity, error) {
+	call := plugin.NewDriverCall(expandVolumeCmd)
+	call.AppendSpec(spec, plugin.host, nil)
+
+	devicePath, err := plugin.getDeviceMountPath(spec)
+	if err != nil {
+		return newSize, err
+	}
+	call.Append(devicePath)
+	call.Append(strconv.FormatInt(newSize.Value(), 10))
+	call.Append(strconv.FormatInt(oldSize.Value(), 10))
+
+	_, err = call.Run()
+	return newSize, err
+}
+
+func (plugin *flexVolumePlugin) RequiresFSResize() bool {
+	return plugin.capabilities.RequiresFSResize
+}
+
+func (plugin *flexVolumePlugin) ExpandFS(spec *volume.Spec, devicePath, deviceMountPath string) error {
+	// This method is called after we spec.PersistentVolume.Spec.Capacity
+	// has been updated to the new size. The underlying driver thus sees
+	// the _new_ (requested) size and can find out the _current_ size from
+	// its underlying storage implementation
+
+	call := plugin.NewDriverCall(expandFSCmd)
+	call.AppendSpec(spec, plugin.host, nil)
+	call.Append(devicePath)
+	call.Append(deviceMountPath)
+
+	_, err := call.Run()
```

Comment:
Same here, use a default function and log it. All other calls do it today.

Reply:
Sure

Reply:
Also, for expandFS, we should use the default EXT/XFS expander if the call is not supported. Some plugins wouldn't want to implement expandFS and we can use the default EXT/XFS expander for them.